src/sql/parts/dashboard/widgets/insights/views/charts/chartInsight.component.ts

```diff
@@ -129,7 +134,14 @@ export abstract class ChartInsight extends Disposable implements IInsightsView {
 		// hence it's easier to not render until ready
 		this.options = mixin(this.options, { maintainAspectRatio: false });
 		this._hasInit = true;
-		this._changeRef.detectChanges();
+		this._hasError = false;
+		try {
+			this._changeRef.detectChanges();
```

Comment:
What is the instance when this errors? Generally if this is happening there is greater angular logic we are not doing correctly. So just wrapping it in a try catch might fix this narrow problem, but there will probably be other related problems.

Reply:
The error comes from within ng2-charts code based on trying to use data that isn't in the expected format. Unless we want to manually verify the data format for each chart type (might be a good idea but would take more time - do you know where that format is documented?) then I'm not sure how else we'd catch it.

Here's an example stack trace:
```
/Users/mairvine/code/sqlopsstudio/node_modules/zone.js/dist/zone-node.js:2251   ERR ng-charts configuration error,
      data or datasets field are required to render char scatter: Error: ng-charts configuration error,
      data or datasets field are required to render char scatter
    at BaseChartDirective.getDatasets (/Users/mairvine/code/sqlopsstudio/node_modules/ng2-charts/charts/charts.js:129:19)
    at BaseChartDirective.getChartBuilder (/Users/mairvine/code/sqlopsstudio/node_modules/ng2-charts/charts/charts.js:57:29)
    at BaseChartDirective.refresh (/Users/mairvine/code/sqlopsstudio/node_modules/ng2-charts/charts/charts.js:139:27)
    at BaseChartDirective.ngOnInit (/Users/mairvine/code/sqlopsstudio/node_modules/ng2-charts/charts/charts.js:28:18)
    at checkAndUpdateDirectiveInline (/Users/mairvine/code/sqlopsstudio/node_modules/@angular/core/bundles/core.umd.js:10822:19)
    at checkAndUpdateNodeInline (/Users/mairvine/code/sqlopsstudio/node_modules/@angular/core/bundles/core.umd.js:12245:17)
    at checkAndUpdateNode (/Users/mairvine/code/sqlopsstudio/node_modules/@angular/core/bundles/core.umd.js:12184:16)
    at debugCheckAndUpdateNode (/Users/mairvine/code/sqlopsstudio/node_modules/@angular/core/bundles/core.umd.js:12887:59)
    at debugCheckDirectivesFn (/Users/mairvine/code/sqlopsstudio/node_modules/@angular/core/bundles/core.umd.js:12828:13)
    at Object.eval [as updateDirectives] (ng:///QueryOutputModule/ScatterChart.ngfactory.js:61:5)
```

Reply:
All the different charts are here http://www.chartjs.org/docs/latest/charts/ we probably should verify all the data and try to be smart about letting the user know why we can't render their chart. 

I attempted to add validation here https://github.com/Microsoft/sqlopsstudio/blob/master/src/sql/parts/dashboard/widgets/insights/views/charts/chartInsight.component.ts#L326 but clearly I didn't catch everything.

Reply:
@anthonydresser that's probably a bigger change than we want to take on right before the release. Would you be ok with merging this change as-is and then we can make those improvements later? As it stands this is a good user experience improvement for this release